app/src/main/java/org/mozilla/fenix/library/history/HistoryView.kt

```diff
@@ -31,8 +33,25 @@ class HistoryView(
     var mode: HistoryFragmentState.Mode = HistoryFragmentState.Mode.Normal
         private set
 
-    val historyAdapter = HistoryAdapter(interactor)
+    val historyAdapter = HistoryAdapter(interactor).apply {
+        addLoadStateListener {
+            // first call will always have itemCount == 0, but we want to keep adapterItemCount
+            // as null until we can distinguish an empty list from populated, so updateEmptyState()
+            // could work correctly
+            if (itemCount != 0) {
+                adapterItemCount = itemCount
+            }
+            if (it.source.refresh is LoadState.NotLoading &&
```

Comment:
Can this be an `else if`? Otherwise, can you add a new line before this to separate the 2 if statement blocks so that it would be easier to read large blocks of code separately.